File: sdks/typescript/src/resources/categories.ts
import { AxiosInstance } from 'axios';
import { PaginatedResponse } from '../types';

export interface Category {
    id: string;
    name: string;
    description?: string;
    parent?: string;
    created_at: string;
    updated_at: string;
}

export class Categories {
    constructor(private client: AxiosInstance) { }

    async list(page: number = 1, pageSize: number = 20): Promise<PaginatedResponse<Category>> {
        const response = await this.client.get('/categories/', {
            params: { page, page_size: pageSize }
        });
        return response.data;
    }

    async get(categoryId: string): Promise<Category> {
        const response = await this.client.get(`/categories/${categoryId}/`);
        return response.data;
    }

    async create(data: Partial<Category>): Promise<Category> {
        const response = await this.client.post('/categories/', data);
        return response.data;
    }

    async update(categoryId: string, data: Partial<Category>): Promise<Category> {
        const response = await this.client.patch(`/categories/${categoryId}/`, data);
        return response.data;
    }

    async delete(categoryId: string): Promise<void> {
        await this.client.delete(`/categories/${categoryId}/`);
    }
}

File: sdks/rust/src/resources/categories.rs
use crate::models::{PaginatedResponse, Category};
use crate::Puxbay;
use crate::PuxbayError;
use reqwest::Method;

pub struct CategoriesResource<'a> {
    pub(crate) client: &'a Puxbay,
}

impl<'a> CategoriesResource<'a> {
    pub fn new(client: &'a Puxbay) -> Self {
        Self { client }
    }

    pub async fn list(&self, page: i32) -> Result<PaginatedResponse<Category>, PuxbayError> {
        self.client.request(
            Method::GET,
            &format!("categories/?page={}", page),
            None::<()>,
        ).await
    }

    pub async fn get(&self, category_id: &str) -> Result<Category, PuxbayError> {
        self.client.request(
            Method::GET,
            &format!("categories/{}/", category_id),
            None::<()>,
        ).await
    }

    pub async fn create(&self, category: &Category) -> Result<Category, PuxbayError> {
        self.client.request(
            Method::POST,
            "categories/",
            Some(category),
        ).await
    }

    pub async fn update(&self, category_id: &str, category: &Category) -> Result<Category, PuxbayError> {
        self.client.request(
            Method::PATCH,
            &format!("categories/{}/", category_id),
            Some(category),
        ).await
    }

    pub async fn delete(&self, category_id: &str) -> Result<(), PuxbayError> {
        self.client.request::<serde_json::Value>(
            Method::DELETE,
            &format!("categories/{}/", category_id),
            None::<()>,
        ).await?;
        Ok(())
    }
}
